### DISCUSSION TITLE
Help understanding differences between "BreakBoundaryOnSubdomainGenerator" and "BreakMeshByBlockGenerator"

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#25624] [Q&A General] [answered]
Hi all, as posted in #24189, the distributed mesh option for "BreakMeshByBlockGenerator" is not available. I'm seeking for other option ("BreakBoundaryOnSubdomainGenerator" + "SideSetsBetweenSubdomainsGenerator") to achieve similar things as "BreakMeshByBlockGenerator"  but allow distributed mesh option. I have a hard time understanding the differences, for example, I created mesh using both ways:

method 1:
```
[Mesh]
    [./msh]
        type = GeneratedMeshGenerator
        dim = 2
        nx = 201
        ny = 200
        xmin = -10050
        xmax = 10050
        ymin = -10000
        ymax = 10000
    [../]
    [./new_block]
        type = ParsedSubdomainMeshGenerator
        input = msh
        combinatorial_geometry = 'y<0'
        block_id = 1
    []
    #add "Block0_Block1" and "Block1_Block0" interfaces
    [./split]
        type = BreakMeshByBlockGenerator
        input = new_block
        split_interface = true
        add_interface_on_two_sides = true
    []
[]
```

method 2:
```
[Mesh]
    [generated1]
        type = GeneratedMeshGenerator
        dim = 2
        nx = 201
        ny = 200
        xmin = -10050
        xmax = 10050
        ymin = -10000
        ymax = 10000
    []
    [./new_block]
        type = ParsedSubdomainMeshGenerator
        input = generated1
        combinatorial_geometry = 'y<0'
        block_id = 1
    []
    [break_boundary]
        type = BreakBoundaryOnSubdomainGenerator
        input = new_block
    []
    [interface]
        type = SideSetsBetweenSubdomainsGenerator
        input = break_boundary
        primary_block = 0
        paired_block = 1
        new_boundary = 'Block0_Block1'
    []
    [interface2]
        type = SideSetsBetweenSubdomainsGenerator
        input = interface
        primary_block = 1
        paired_block = 0
        new_boundary = 'Block1_Block0'
    []
[]
```
And would like to apply some boundary conditions along the two interfaces using **Explicit Time Stepping**. 

```
[BCs]
    #assign displacement boundary condition
    [./matchval_primary_x]
        type = MatchedValueBC
        variable = disp_x
        v = disp_plus_scaled_x
        boundary = 'Block0_Block1'
    []
    [./matchval_secondary_x]
        type = MatchedValueBC
        variable = disp_x
        v = disp_minus_scaled_x
        boundary = 'Block1_Block0'
    []
    [./matchval_primary_y]
        type = MatchedValueBC
        variable = disp_y
        v = disp_plus_scaled_y
        boundary = 'Block0_Block1'
    []
    [./matchval_secondary_y]
        type = MatchedValueBC
        variable = disp_y
        v = disp_minus_scaled_y
        boundary = 'Block1_Block0'
    []
[]
```

As I tested, there are differences in the interfaces created by the two methods , but it troubles me to fully understand the discrepancies. So thank advance for anyone who is willing to help!

**Comment by GiudGiud** [comment]
hello

sidesets are oriented. A side in a sideset is an element and a side index. So it's oriented away from that element. 

If you want the opposite orientation, you take the neigjbor element and the side index for the first element
That's why there's two possible definitions, with two different normals, what "looks" the same in the exodus output

**Reply by chunhuizhao478** [reply]
Hi Guid, thanks for your reply. I think my question is a bit misleading, I understand the orientation of block0_block1 and block1_block0 are different, my question is do I get the same two interfaces by using breakmeshbyblock or breakboundyonsudomain, so the difference between the two generator is what I want to know, thanks!

**Reply by GiudGiud** [reply]
in BMBBYMG:
```
A new sideset identifying the new interface is added, and it is always linked to elements belonging to blocks with the lower ID.
```
so just make sure the primary block in the BreakMeshByBoundary always is the block with the lowest ID

**Comment by chunhuizhao478** [comment, marked as answer]
I think I find a answer to my question in the earlier post:

https://github.com/idaholab/moose/discussions/23955#discussioncomment-5522867

BreakMeshByBlockGenerator : split the mesh 
BreakBoundaryOnSubdomainGenerator : create a new sideset
